python/packages/autogen-agentchat/src/autogen_agentchat/teams/_utils.py

```diff
@@ -0,0 +1,89 @@
+from datetime import datetime
+import json
+import logging
+import sys
+from typing import Optional, Union, List, Dict, Any, Sequence
+from dataclasses import asdict, is_dataclass
+
+from .group_chat._events import ContentPublishEvent
+from ..agents import ChatMessage, TextMessage, MultiModalMessage, ToolCallMessage, ToolCallResultMessage, StopMessage
+from autogen_core.components import FunctionCall, Image
+from autogen_core.components.models import FunctionExecutionResult
+
+ContentType = Union[str, List[Union[str, Image]],
+                    List[FunctionCall], List[FunctionExecutionResult]]
+
+
+class AgentChatLogHandler(logging.Handler):
+    def __init__(self, filename: Optional[str] = None) -> None:
+        super().__init__()
+        self.filename = filename
+        self.file_handler: Optional[logging.FileHandler] = None
+        if filename:
+            self.file_handler = logging.FileHandler(filename)
+
+    def emit(self, record: logging.LogRecord) -> None:
+        try:
+            ts = datetime.fromtimestamp(record.created).isoformat()
+            if isinstance(record.msg, ContentPublishEvent):
+                console_message = (
+                    f"\n{'-'*75} \n"
+                    f"\033[91m[{ts}], {record.msg.agent_message.source}:\033[0m\n"
+                    f"\n{self.serialize_content(record.msg.agent_message.content)}"
+                )
+                # print , flush true
+                sys.stdout.write(console_message)
+
+                log_entry = json.dumps(
+                    {
+                        "timestamp": ts,
+                        "source": record.msg.agent_message.source,
+                        "message": self.serialize_content(record.msg.agent_message.content),
+                        "type": "OrchestrationEvent",
+                    },
+                    default=self.json_serializer,
+                )
+
+                if self.file_handler:
```

Comment:
Seems like there are two separate output streams here, perhaps it deserves to be two separate log handler classes instead of one? We can add them both by default, or just the console one by default, and have user add it when needed.

Reply:
This way we keep consistent with Python's native logging API so no need to learn new things.

Reply:
All good ideas.
I made console default for now in `_base_team`
```python
logger = logging.getLogger(EVENT_LOGGER_NAME)
logger.setLevel(logging.INFO)
console_handler = ConsoleLogHandler()
logger.addHandler(console_handler)
```